xml/System.Diagnostics/DiagnosticSource.xml

```diff
@@ -188,11 +193,14 @@
         <Parameter Name="value" Type="System.Object" />
       </Parameters>
       <Docs>
-        <param name="name">To be added.</param>
-        <param name="value">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="name">The name of the event being written.</param>
+        <param name="value">An object that represent the value being passed as a payload for the event. This is often an anonymous type which contains several sub-values.</param>
+        <summary>Write is a generic way of logging complex payloads. Each notification is given a name, which identifies it as well as a object (typically an anonymous type) that gives the information to pass to the notification, which is arbitrary.  
+The <paramref name="name" /> should be short (so don't use fully qualified names unless you have to to avoid ambiguity), but you want the <paramref name="name" /> to be globally unique. Typically, your componentName.eventName, where componentName and eventName are strings less than 10 characters, are a good compromise.
+Notification names should not have '.' in them because component names have dots and for them both to have dots would lead to ambiguity. The suggestion is to use _ instead.
+It is assumed that listeners will use string prefixing to filter groups, thus having hierarchy in component names is good.</summary>
         <remarks>To be added.</remarks>
```

Comment:
```suggestion
        <remarks>  
          <format type="text/markdown"><![CDATA[  

Each notification is given a name that identifies it, as well as an object (typically an anonymous type) that provides arbitrary information to pass to the notification.  

`name` should be short. Don't use a fully qualified name unless you have to to avoid ambiguity, since `name` must be globally unique. Typically, *componentName.eventName*, where *componentName* and *eventName* are strings less than 10 characters, are a good compromise.

Notification names should not have `.` in them because component names have dots, and for them both to have dots leads to ambiguity. We suggest that you use `_` instead.

Assume that listeners will use string prefixing to filter groups. Therefore, having a hierarchy of component names is a good practice.

         ]]></format>
         </remarks>
```